zerver/tests/test_embedded_bot_system.py

```diff
@@ -52,6 +53,31 @@ def test_stream_message_to_embedded_bot(self) -> None:
         display_recipient = get_display_recipient(last_message.recipient)
         self.assertEqual(display_recipient, "Denmark")
 
+    def test_embedded_bot_react_to_stream_message(self) -> None:
+        assert self.bot_profile is not None
+        self.send_stream_message(
+            self.user_profile,
+            "Denmark",
+            content=f"@**{self.bot_profile.full_name}** foo",
+            topic_name="bar",
+        )
+        user_message = self.get_second_to_last_message()
+        self.assertTrue(
+            Reaction.objects.filter(
+                message=user_message,
+            ).exists()
+        )
+
+    def test_embedded_bot_react_to_pm(self) -> None:
```

Comment:
A test for Failing reaction will help here too.

Reply:
Actually, this shouldn't matter whether it's pm or stream message, as the exceptions that we are interested in occur before the reaction is actually added in `do_add_reaction`.